### ISSUE TITLE
Shape mismatch on mnist tutorial example

### ISSUE THREAD

**Issue by sunny721330** [tflearn/tflearn#717]
I downloaded mnist tutorial example, it runs well on my computer (tensorflow cpu), but when I try to run it on the cluster with tensorflow gpu version, I got the following mistake, basically it says there is a shape mismatch somewhere:

Caused by op 'save/Assign_8', defined at:
  File "lenet.py", line 164, in <module>
    tf.app.run()
  File "/cnn/lib/python3.5/site-packages/tensorflow/python/platform/app.py", line 44, in run
    _sys.exit(main(_sys.argv[:1] + flags_passthrough))
  File "lenet.py", line 148, in main
    monitors=[logging_hook])
  File "/cnn/lib/python3.5/site-packages/tensorflow/contrib/learn/python/learn/estimators/estimator.py", line 1353, in fit
    monitors=all_monitors)
  File "/cnn/lib/python3.5/site-packages/tensorflow/python/util/deprecation.py", line 280, in new_func
    return func(*args, **kwargs)
  File "/cnn/lib/python3.5/site-packages/tensorflow/contrib/learn/python/learn/estimators/estimator.py", line 426, in fit
    loss = self._train_model(input_fn=input_fn, hooks=hooks)
  File "/1/cnn/lib/python3.5/site-packages/tensorflow/contrib/learn/python/learn/estimators/estimator.py", line 981, in _train_model
    config=self.config.tf_config) as mon_sess:
  File "/1/cnn/lib/python3.5/site-packages/tensorflow/python/training/monitored_session.py", line 315, in MonitoredTrainingSession
    return MonitoredSession(session_creator=session_creator, hooks=all_hooks)
  File "/1/cnn/lib/python3.5/site-packages/tensorflow/python/training/monitored_session.py", line 601, in __init__
    session_creator, hooks, should_recover=True)
  File "/1/cnn/lib/python3.5/site-packages/tensorflow/python/training/monitored_session.py", line 434, in __init__
    self._sess = _RecoverableSession(self._coordinated_creator)
  File "/cnn/lib/python3.5/site-packages/tensorflow/python/training/monitored_session.py", line 767, in __init__
    _WrappedSession.__init__(self, self._create_session())
  File "/cnn/lib/python3.5/site-packages/tensorflow/python/training/monitored_session.py", line 772, in _create_session
    return self._sess_creator.create_session()
  File "/cnn/lib/python3.5/site-packages/tensorflow/python/training/monitored_session.py", line 494, in create_session
    self.tf_sess = self._session_creator.create_session()
  File "/cnn/lib/python3.5/site-packages/tensorflow/python/training/monitored_session.py", line 366, in create_session
    self._scaffold.finalize()
  File "/cnn/lib/python3.5/site-packages/tensorflow/python/training/monitored_session.py", line 183, in finalize
    self._saver.build()
  File "/cnn/lib/python3.5/site-packages/tensorflow/python/training/saver.py", line 1070, in build
    restore_sequentially=self._restore_sequentially)
  File "/cnn/lib/python3.5/site-packages/tensorflow/python/training/saver.py", line 671, in build
    restore_sequentially, reshape)
  File "/cnn/lib/python3.5/site-packages/tensorflow/python/training/saver.py", line 445, in _AddShardedRestoreOps
    name="restore_shard"))
  File "/cnn/lib/python3.5/site-packages/tensorflow/python/training/saver.py", line 414, in _AddRestoreOps
    assign_ops.append(saveable.restore(tensors, shapes))
  File "/cnn/lib/python3.5/site-packages/tensorflow/python/training/saver.py", line 155, in restore
    self.op.get_shape().is_fully_defined())
  File "/cnn/lib/python3.5/site-packages/tensorflow/python/ops/gen_state_ops.py", line 47, in assign
    use_locking=use_locking, name=name)
  File "/cnn/lib/python3.5/site-packages/tensorflow/python/framework/op_def_library.py", line 763, in apply_op
    op_def=op_def)
  File "/cnn/lib/python3.5/site-packages/tensorflow/python/framework/ops.py", line 2327, in create_op
    original_op=self._default_original_op, op_def=op_def)
  File "//cnn/lib/python3.5/site-packages/tensorflow/python/framework/ops.py", line 1226, in __init__
    self._traceback = _extract_stack()

InvalidArgumentError (see above for traceback): Assign requires shapes of both tensors to match. **lhs shape= [1024,10] rhs shape= [1024,2]**
     [[Node: save/Assign_8 = Assign[T=DT_FLOAT, _class=["loc:@dense_1/kernel"], use_locking=true, validate_shape=true, _device="/job:localhost/replica:0/task:0/cpu:0"](dense_1/kernel, save/RestoreV2_8)]]

**Comment by aymericdamien** [comment]
It seems to have an issue for the latest fully connected layer, it should have 10 units (for the 10 classes) but somehow it only have 2.
